### PR TITLE
feat(port-crawler-to-cli): Add simulator page processor

### PR DESCRIPTION
**Pull request by AhmedAbdoOrtiga**
#### Description of changes
This PR adds another page processor that simulates user click on elements that match to the specified selectors, if the simulate arg is true, the simuator page will run using the passed selectors in the args, the default is button.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/crawler/src/crawler/crawler-engine.spec.ts [hunk 1]
```diff
@@ -82,14 +96,15 @@ describe(CrawlerEngine, () => {
     });
 
     afterEach(() => {
-        pageProcessorFactoryMock.verifyAll();
+        // pageProcessorFactoryMock.verifyAll();
+        // crawlerFactoryMock.verifyAll();
+        // puppeteerCrawlerMock.verifyAll();
```
Comment (pownkel) [inline packages/crawler/src/crawler/crawler-engine.spec.ts:101]: Why are these commented out?

File: .vscode/launch.json [hunk 2]
```diff
@@ -5,15 +5,20 @@
     "version": "0.2.0",
     "configurations": [
         {
+            "name": "Debug current test file in VSCode",
```
Comment (lamaks) [inline .vscode/launch.json:8]: This is global to solution change. Does this work across all packages?

File: packages/crawler/src/apify-resources/resource-creator.ts [hunk 3]
```diff
@@ -2,13 +2,12 @@
 // Licensed under the MIT License.
 import Apify from 'apify';
 
-export interface CrawlerFactory {
+export interface ResourceCreator {
```
Comment (lamaks) [inline packages/crawler/src/apify-resources/resource-creator.ts:5]: Too generic name, rename to ApifyResourceCreator?

File: packages/crawler/src/index.ts [hunk 4]
```diff
@@ -4,4 +4,58 @@
...
+    const crawlerEngine: CrawlerEngine = new CrawlerEngine();
+
+    await crawlerEngine.start({
+        baseUrl: args.url,
+        simulate: args.simulate,
+        selectors: args.selectors,
+        localOutputDir: args.output,
+    });
+})().catch((error) => {
+    console.log('Exception: ', error);
```
Comment (lamaks) [inline packages/crawler/src/index.ts:59]: serialize error with `System.serializeError`

File: .vscode/launch.json (hunk 2 above)
Comment (AhmedAbdoOrtiga) [reply .vscode/launch.json:8]: It works when you debug unit test in vscode by clicking f5 for any unit test file.

File: packages/crawler/src/apify-resources/resource-creator.ts (hunk 3 above)
Comment (AhmedAbdoOrtiga) [reply packages/crawler/src/apify-resources/resource-creator.ts:5]: This is the interface, that's why it's generic, the implementation of it in the same file is ApifyResourceCreator which uses apify resources, later we can use something like MemoryResourceCreator for using the memory as a resource

File: packages/crawler/src/crawler/crawler-engine.spec.ts (hunk 1 above)
Comment (AhmedAbdoOrtiga) [reply packages/crawler/src/crawler/crawler-engine.spec.ts]: fixed, was working on it

File: packages/crawler/src/index.ts (hunk 4 above)
Comment (AhmedAbdoOrtiga) [reply packages/crawler/src/index.ts]: fixed

**Pull request merged**